src/check_release.rs

```diff
@@ -137,85 +137,138 @@ fn print_triggered_lint(
-                let indented_serde = serde_pretty
-                    .split('\n')
-                    .map(|line| format!("    {line}"))
-                    .join("\n");
+    // Use appropriate log level based on lint severity
+    match lint_result.effective_lint_level {
+        LintLevel::Deny => {
+            config.log_error(|config| {
+                if let Some(ref_link) = semver_query.reference_link.as_deref() {
+                    writeln!(config.stdout(), "{}Description:{}\n{}\n{:>12} {}\n{:>12} https://github.com/obi1kenobi/cargo-semver-checks/tree/v{}/src/lints/{}.ron\n",
+                        Style::new().bold(), Reset,
+                        &semver_query.error_message,
+                        "ref:",
+                        ref_link,
+                        "impl:",
+                        crate_version!(),
+                        semver_query.id,
+                    )?;
+                } else {
+                    writeln!(
+                        config.stdout(),
+                        "{}Description:{}\n{}\n{:>12} https://github.com/obi1kenobi/cargo-semver-checks/tree/v{}/src/lints/{}.ron",
+                        Style::new().bold(),
+                        Reset,
+                        &semver_query.error_message,
+                        "impl:",
+                        crate_version!(),
+                        semver_query.id,
+                    )?;
+                }
                 writeln!(
                     config.stdout(),
-                    "\tlint rule output values:\n{indented_serde}"
+                    "{}Failed in:{}",
+                    Style::new().bold(),
+                    Reset
                 )?;
                 Ok(())
             })?;
-        } else {
-            config.log_info(|config| {
+        }
+        LintLevel::Warn => {
+            config.log_warn(|config| {
+                if let Some(ref_link) = semver_query.reference_link.as_deref() {
+                    writeln!(config.stdout(), "{}Description:{}\n{}\n{:>12} {}\n{:>12} https://github.com/obi1kenobi/cargo-semver-checks/tree/v{}/src/lints/{}.ron\n",
+                        Style::new().bold(), Reset,
+                        &semver_query.error_message,
+                        "ref:",
+                        ref_link,
+                        "impl:",
+                        crate_version!(),
+                        semver_query.id,
+                    )?;
+                } else {
+                    writeln!(
+                        config.stdout(),
+                        "{}Description:{}\n{}\n{:>12} https://github.com/obi1kenobi/cargo-semver-checks/tree/v{}/src/lints/{}.ron",
+                        Style::new().bold(),
+                        Reset,
+                        &semver_query.error_message,
+                        "impl:",
+                        crate_version!(),
+                        semver_query.id,
+                    )?;
+                }
```

Comment:
Can we avoid the duplication between the `log_error` and `log_warn` cases, both here and below? This feels like it would regress maintainability a decent bit, since this code is not trivial.